src/Mod/Draft/draftobjects/dimension.py

```diff
@@ -0,0 +1,370 @@
+            if len(obj.LinkedGeometry) == 1:
+                lobj = obj.LinkedGeometry[0][0]
+                lsub = obj.LinkedGeometry[0][1]
+                if len(lsub) == 1:
+                    if "Edge" in lsub[0]:
+                        n = int(lsub[0][4:])-1
+                        edge = lobj.Shape.Edges[n]
+                        if DraftGeomUtils.geomType(edge) == "Line":
+                            obj.Start = edge.Vertexes[0].Point
+                            obj.End = edge.Vertexes[-1].Point
+                        elif DraftGeomUtils.geomType(edge) == "Circle":
+                            c = edge.Curve.Center
+                            r = edge.Curve.Radius
+                            a = edge.Curve.Axis
+                            ray = obj.Dimline.sub(c).projectToPlane(App.Vector(0,0,0),a)
+                            if (ray.Length == 0):
+                                ray = a.cross(App.Vector(1,0,0))
+                                if (ray.Length == 0):
+                                    ray = a.cross(App.Vector(0,1,0))
+                            ray = DraftVecUtils.scaleTo(ray,r)
+                            if hasattr(obj,"Diameter"):
+                                if obj.Diameter:
+                                    obj.Start = c.add(ray.negative())
+                                    obj.End = c.add(ray)
+                                else:
+                                    obj.Start = c
+                                    obj.End = c.add(ray)
+                elif len(lsub) == 2:
+                    if ("Vertex" in lsub[0]) and ("Vertex" in lsub[1]):
+                        n1 = int(lsub[0][6:])-1
+                        n2 = int(lsub[1][6:])-1
+                        obj.Start = lobj.Shape.Vertexes[n1].Point
+                        obj.End = lobj.Shape.Vertexes[n2].Point
+            elif len(obj.LinkedGeometry) == 2:
+                lobj1 = obj.LinkedGeometry[0][0]
+                lobj2 = obj.LinkedGeometry[1][0]
+                lsub1 = obj.LinkedGeometry[0][1]
+                lsub2 = obj.LinkedGeometry[1][1]
+                if (len(lsub1) == 1) and (len(lsub2) == 1):
+                    if ("Vertex" in lsub1[0]) and ("Vertex" in lsub2[1]):
+                        n1 = int(lsub1[0][6:])-1
+                        n2 = int(lsub2[0][6:])-1
+                        obj.Start = lobj1.Shape.Vertexes[n1].Point
+                        obj.End = lobj2.Shape.Vertexes[n2].Point
+        # set the distance property
+        total_len = (obj.Start.sub(obj.End)).Length
+        if round(obj.Distance.Value, utils.precision()) != round(total_len, utils.precision()):
+            obj.Distance = total_len
+        if App.GuiUp:
+            if obj.ViewObject:
+                obj.ViewObject.update()
+
+
+
+class AngularDimension(DimensionBase):
+    """
+    The Draft AngularDimension object
+    """
+
+    def __init__(self, obj):
+
+        super().__init__(obj,"AngularDimension")
```

Comment:
This is failing the Travis tests for Python 2. In this version the syntax is `super(DimensionBase, self)`. It probably should be corrected if this is going to be released in Python 2.

Reply:
It's just good if I substitute it?

Reply:
An easy way to fix this (not sure about how pythonic that is...) woud be instead of calling super(), calling explicitely DimensionBase.__init__(obj,"AngularDimension") I think that works equally for both python versions